include/swift/Basic/Located.h

```diff
@@ -22,15 +21,27 @@
 
 namespace swift {
 
+/// A currency type for keeping track of items which were found in the source code.
+///
+/// Several parts of the compiler need to keep track of a `SourceLoc` corresponding to an item, in case
+/// they need to report some diagnostics later. For example, the ClangImporter needs to keep track
+/// of where imports were originally written. Located makes it easy to do so while making the code
+/// more readable, compared to using `std::pair`.
 template<typename T>
 struct Located {
 
+  /// The main item whose source location is being tracked.
   T item;
 
+  /// The original source location from which the item was parsed.
   SourceLoc loc;
 
+  Located() {}
+
+  Located(T item, SourceLoc loc): item(item), loc(loc) {}
+
   template<typename U>
-  friend bool operator ==(const Located<U> lhs, const Located<U> rhs) {
+  friend bool operator ==(const Located<U>& lhs, const Located<U>& rhs) {
```

Comment:
@CodaFi's comment was probably about this line, which should have `const Located<U> &lhs` instead of `const Located<U>& lhs` and similarly for `rhs`.

Reply:
Fixed.